vignettes/datatable-joins.Rmd

```diff
@@ -529,6 +529,14 @@ A non-equi join is a type of join where the condition for matching rows is based
 - Finding the nearest match.
 - Comparing ranges of values between tables.
 
+### Key Constraints
+- **Column Requirements**:
+  - Only column *names* (not expressions) can be used with binary operators
+  - Left operand must be from `x` (left table)
+  - Right operand must be from `i` (right table)
+- **Naming Behavior**:
+  - Joined columns inherit names from `x` but contain values from `i`
+
```

Comment:
This is not requested in the linked issue, nor described+motivated in the PR, so I'm dropping it. I have some quibbles with whether it should be here / it needs some refinement. Better to just merge the fix as is for now.

```suggestion
```